src/item_factory.cpp

```diff
@@ -872,7 +872,6 @@ void Item_factory::load( islot_container &slot, JsonObject &jo )
     slot.seals = jo.get_bool( "seals", false );
     slot.watertight = jo.get_bool( "watertight", false );
     slot.preserves = jo.get_bool( "preserves", false );
-    slot.rigid = jo.get_bool( "rigid", false );
```

Comment:
Now the "rigid" flag in the container slot data in JSON will be completely ignored. If you change the way the JSON data is read, you should update the JSON (including the contents of mods) to match the new syntax.

- Move the member from the container slot into the common item data (for example needed for the jerrypack), the "rigid" member inside the container slot is not used anymore.

- Fix the containers that don't have an explicit "rigid" value set. The PR changes the default from `false` (for containers) to `true`, so all containers that don't have an explicit value set (and were `rigid = false`) are now `rigid = true`.

Test this with a few containers. I have a "large waterskin of water (20)" that has volume 3, when emptied, it still has volume 3.